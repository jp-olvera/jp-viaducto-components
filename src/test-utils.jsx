import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
