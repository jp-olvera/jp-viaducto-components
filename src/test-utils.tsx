import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
