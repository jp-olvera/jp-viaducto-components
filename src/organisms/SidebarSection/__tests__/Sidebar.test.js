import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '../../../test-utils';
import { SidebarSection } from '..';

jest.mock('../../../cells/Dropdown/sorting.svg', () => null);

describe('<SidebarSection/>', () => {
  let items = [
    {
      label: 'title',
      url: 'url',
    },
    {
      label: 'section',
      url: 'google',
    },
    {
      label: 'menu',
      url: 'youtube',
    },
  ];
  test('should be visible', () => {
    render(
      <SidebarSection
        items={items}
        separator={true}
        title={'Comida'}
        isDropdown={true}
        isMenu={false}
        lead={true}
      />
    );
    expect(screen.getByText('Comida')).toBeVisible();
  });
  test('should render option of SidebarSection', () => {
    render(
      <SidebarSection
        items={items}
        separator={true}
        title={'Comida'}
        isDropdown={false}
        isMenu={false}
        lead={true}
      />
    );
    expect(screen.getByText('menu')).toBeVisible();
  });
  test('should render option after dropdown click', () => {
    render(
      <SidebarSection
        items={items}
        separator={true}
        title={'Comida'}
        isDropdown={true}
        isMenu={false}
        lead={true}
      />
    );
    expect(screen.queryByText('menu')).not.toBeInTheDocument();
    fireEvent.click(screen.getByText('Comida'));
    expect(screen.getByText('menu')).toBeVisible();
  });
  test('should render option after menu click', () => {
    render(
      <SidebarSection
        items={items}
        separator={true}
        title={'Comida'}
        isDropdown={false}
        isMenu={true}
        lead={true}
      />
    );
    expect(screen.queryByText('menu')).not.toBeInTheDocument();
    fireEvent.click(screen.getByText('Comida'));
    expect(screen.getByText('menu')).toBeVisible();
  });
});
