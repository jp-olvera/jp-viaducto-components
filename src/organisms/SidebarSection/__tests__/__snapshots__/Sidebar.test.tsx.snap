// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarSection/> should render with default props 1`] = `
<div>
  <div
    class="StyledSidebarSection-sc-12e8js3-0 hSnxND"
  >
    <div
      class=""
    >
      <div
        class="StyledSidebarSection__StyledMenuTitle-sc-12e8js3-3 dLwqie"
        tabindex="0"
        title=""
        type=""
      >
        <span
          style="display: flex; align-items: center;"
        />
        <span
          style="align-items: center; width: 100%; overflow: hidden; white-space: nowrap; text-overflow: ellipsis;"
        />
      </div>
      <div />
    </div>
  </div>
</div>
`;
